packages/generic-utils/src/emitter/__tests__/emitter.test.ts

```diff
@@ -71,4 +71,36 @@ describe(Emitter, () => {
 
     expect(fn).toHaveBeenCalledTimes(2)
   })
+
+  it('has a default max listeners of 10', () => {
+    const em = new Emitter()
+    expect(em.maxListeners).toBe(10)
+  })
+
+  it('should warn if possible memory leak', () => {
+    const fn = jest.fn()
+    const warnSpy = jest.spyOn(console, 'warn').mockImplementation(() => {})
+    const em = new Emitter({ maxListeners: 3 })
+    em.on('test', fn)
+    em.on('test', fn)
+    em.on('test', fn)
+    expect(warnSpy).not.toHaveBeenCalled()
+    // call on 4th
+    em.on('test', fn)
+    expect(warnSpy).toHaveBeenCalledTimes(1)
+    // do not call additional times
+    em.on('test', fn)
+    expect(warnSpy).toHaveBeenCalledTimes(1)
+  })
+
+  it('has no warning if listener limit is set to 0', () => {
+    const fn = jest.fn()
+    const warnSpy = jest.spyOn(console, 'warn').mockImplementation(() => {})
+    const em = new Emitter({ maxListeners: 0 })
+    expect(em.maxListeners).toBe(0)
+    em.on('test', fn)
+    expect(warnSpy).not.toHaveBeenCalled()
+    em.on('test', fn)
+    expect(warnSpy).not.toHaveBeenCalled()
```

Comment:
nit: would it be worth adding as many listeners as the default (10) to guarantee that's not using the default as `maxListeners`?

Reply:
Sure!